scalding-core/src/main/scala/com/twitter/scalding/typed/Joiner.scala

```diff
@@ -18,36 +18,163 @@ package com.twitter.scalding.typed
 import com.twitter.scalding._
 
 object Joiner extends java.io.Serializable {
-  def toCogroupJoiner2[K, V, U, R](hashJoiner: (K, V, Iterable[U]) => Iterator[R]): (K, Iterator[V], Iterable[U]) => Iterator[R] = {
-    (k: K, itv: Iterator[V], itu: Iterable[U]) =>
-      itv.flatMap { hashJoiner(k, _, itu) }
+
+  type JoinFn[K, V, U, R] = (K, Iterator[V], Iterable[U]) => Iterator[R]
+  type HashJoinFn[K, V, U, R] = (K, V, Iterable[U]) => Iterator[R]
+
+  def toCogroupJoiner2[K, V, U, R](hashJoiner: (K, V, Iterable[U]) => Iterator[R]): JoinFn[K, V, U, R] =
```

Comment:
Can you add a comment on this method? (motivation as much as anything -- not sure why we would normally want to turn a hash join into a JoinFn normally?) 

also I think we can use the `HashJoinFn[K, V, U,R]` type alias as the type of the arg to this method

Reply:
will do these in the follow up, if that's okay.

We never use this method in fact, but I didn't want to remove it in the interest of being a nice guy to users that may be.

Reply:
Yep totes. and makes sense to leave or kill for the obvious reasons to do either